Core API and test support

chrylis opened the pull request.

chrylis committed:
Core API and test support

CocoTheCoder commented on maildrop-spring-boot-starter/src/main/java/module-info.java:
```diff
@@ -0,0 +1,11 @@
+module co.vendorflow.oss.maildrop.test.autoconfigure {
+    /*exports*/ opens co.vendorflow.oss.maildrop.autoconfigure;
```
Is there a reason that `/*exports*/` is commented out and left here?

chrylis replied:
I chose not to do that cleanup because I still need to do a little testing. Figuring out exactly how the module system works is in progress; it seems that neither `exports` nor `opens` is a subset of the other, so I'm determining the precise rule that's required.

CocoTheCoder reviewed (approved):
LGTM — I left one question about a piece of commented out code.

chrylis committed:
Add temporary Jackson support and tests

The Jackson module declared here has been submitted for inclusion
upstream, hopefully in the 2.13 release, and will be eliminated as
soon as that's available. Testing will remain in place to ensure that
round-trip JSON serialization works cleanly for mail messages.
